tests/debug-test.js

```diff
@@ -85,45 +86,26 @@ describe('debug', function () {
       });
 
       var Component = localComp('DisplayName', function () {
-        return React.DOM.text({ children: 'hello' });
+        return textNode('hello');
       });
       render(Component());
     });
 
     it('should log on render when debug with key', function (done) {
-      var localComp = component.withDefaults();
-      localComp.debug(function logger (message) {
-        message.should.contain('foobar');
-        message.should.contain('render');
-        done();
-      });
-
-      var Component = localComp(function () {
-        return React.DOM.text({ children: 'hello' });
-      });
-      render(Component({ key: 'foobar' }));
-    });
+      var mount = global.document.createElement('div');
 
-    it('should log when debug with key for React 0.13 beta', function (done) {
       var localComp = component.withDefaults();
       localComp.debug(function logger (message) {
         message.should.contain('foobar');
         message.should.contain('render');
+        mount = undefined;
         done();
       });
 
-      var Component = localComp({
-        componentWillMount: function () {
-          this._reactInternalInstance = {
-            _currentElement: {
-              key: 'foobar'
-            }
-          };
-        }
-      }, function () {
-        return React.DOM.text({ children: 'hello' });
+      var Component = localComp(function () {
+        return textNode('hello');
       });
-      render(Component());
+      ReactDOM.render(Component('foobar'), mount);
```

Comment:
In React 16, `ReactDOMServer.renderToString` produces a different representation of the component instance. That representation has no reference to `key` anywhere on the public API of the instance. So this test renders to the DOM instead of to string.